packages/vuetify/src/composables/dateFormat.ts

```diff
@@ -0,0 +1,131 @@
+// Composables
+import { useDate } from '@/composables/date/date'
+
+// Utilities
+import { toRef } from 'vue'
+import { propsFactory } from '@/util'
+
+// Types
+import type { Ref } from 'vue'
+
+// Types
+export interface DateFormatProps {
+  inputFormat?: string
+}
+
+class DateFormatSpec {
+  constructor (
+    public readonly order: string, // mdy | dmy | ymd
+    public readonly separator: string // / | - | .
+  ) { }
+
+  get format () {
+    return this.order.split('')
+      .map(sign => `${sign}${sign}`)
+      .join(this.separator)
+      .replace('yy', 'yyyy')
+  }
+
+  static canBeParsed (v: any) {
+    if (typeof v !== 'string') return false
+    const lowercase = v.toLowerCase()
+    return ['y', 'm', 'd'].every(sign => lowercase.includes(sign)) &&
+      ['/', '-', '.'].some(sign => v.includes(sign))
+  }
+
+  static parse (v: string) {
+    if (!DateFormatSpec.canBeParsed(v)) {
+      throw new Error(`[${v}] cannot be parsed into date format specification`)
+    }
+    const order = v.toLowerCase().split('')
+      .filter((c, i, all) => 'dmy'.includes(c) && all.indexOf(c) === i)
+      .join('')
+    const separator = ['/', '-', '.'].find(sign => v.includes(sign))!
+    return new DateFormatSpec(order, separator)
+  }
+}
+
+export const makeDateFormatProps = propsFactory({
+  inputFormat: {
+    type: String,
+    validator: (v: string) => !v || DateFormatSpec.canBeParsed(v),
+  },
+}, 'lazy')
+
+export function useDateFormat (props: DateFormatProps, locale: Ref<string>) {
+  const adapter = useDate()
+
+  function inferFromLocale () {
+    return Intl.DateTimeFormat(locale.value ?? 'en-US', { year: 'numeric', month: '2-digit', day: '2-digit' })
+      .format(adapter.toJsDate(adapter.parseISO('1999-12-07')))
+      .replace(/(07)|(٠٧)|(٢٩)|(۱۶)|(০৭)/, 'dd')
+      .replace(/(12)|(١٢)|(٠٨)|(۰۹)|(১২)/, 'mm')
+      .replace(/(1999)|(2542)|(١٩٩٩)|(١٤٢٠)|(۱۳۷۸)|(১৯৯৯)/, 'yyyy')
+      .replace(/[^ymd\-/.]/g, '')
+      .replace(/\.$/, '')
+  }
```

Comment:
I think we should also validate the infered input format using `canBeParsed` just in case it returns an invalid format and fall back to a hardcoded input-format in case it is invalid

Reply:
In case it is invalid we could do one of the following two things:
1. Fall back to a hardcoded input-format. (like I mentioned above)
2. Use the Date adapter's .isValid and .date to validate and parse the input date. (behaviour we had before input-format was implemented)

Reply:
Lazy naming.. I'm open to suggestions

Reply:
`inferFromLocale` is also good but I also like `generateFromLocale`, `deriveFromLocal`,